### ISSUE TITLE
Selecting language with --locale does not always work

### ISSUE THREAD

**Issue by matsduf** [dotse/zonemaster-cli#46]
The command line tool `zonemaster-cli` has an option `--locale` with which you can
select the language the output is to be written in. Currently English (the default), French, 
Swedish or Danish can be selected.

**Under Ubuntu, Debian or CentOS**

If the environment variable `LANGUAGE` is set, then the `--locale` option might not work. 
See the examples from Ubuntu further down. The work-around is to unset or reset the 
LANGUAGE variable before running the `zonemaster-cli` command, e.g.

```sh
unset LANGUAGE
zonemaster-cli zonemaster.net --locale sv_SE.utf8
```

```sh
LANGUAGE=sv_SE.utf8 zonemaster-cli zonemaster.net
```

**Under FreeBSD**

The `--locale` option does not currently work under FreeBSD. You can use the following 
work-around:

```sh
LC_ALL=sv_SE.utf8 zonemaster-cli zonemaster.net
```

**Examples run from Ubuntu**

```
$ # Variables set with locale:
$ env | grep -i utf-8
LC_ALL=sv_SE.UTF-8
LANG=sv_SE.UTF-8
LANGUAGE=sv_SE.UTF-8
LC_CTYPE=sv_SE.UTF-8

$ # Running default gives Swedish messages:
$ zonemaster-cli iis.se
 Sekund Nivå      Meddelande
======= ====      ==========
   5.16 NOTIS     Namnservern 'i.ns.se' har en IP-adress (194.146.106.22) med en icke-matchande bakåtuppslagning (se1.dnsnode.net.).
   5.16 NOTIS     Namnservern 'i.ns.se' har en IP-adress (2001:67c:1010:5::53) med en icke-matchande bakåtuppslagning (se1.dnsnode.net.).

$ # Selecting French locale with --locale does not help:
$ zonemaster-cli iis.se --locale fr_FR.utf8
 Sekund Nivå      Meddelande
======= ====      ==========
  10.14 NOTIS     Namnservern 'i.ns.se' har en IP-adress (194.146.106.22) med en icke-matchande bakåtuppslagning (se1.dnsnode.net.).
  10.15 NOTIS     Namnservern 'i.ns.se' har en IP-adress (2001:67c:1010:5::53) med en icke-matchande bakåtuppslagning (se1.dnsnode.net.).

$ # Unsetting all variables gives default or English output:
$ (unset LC_ALL; unset LANG; unset LANGUAGE; unset LC_CTYPE; zonemaster-cli iis.se)
Seconds Level     Message
======= ========= =======
   5.21 NOTICE    Nameserver i.ns.se has an IP address (194.146.106.22) with mismatched PTR result (se1.dnsnode.net.).
   5.21 NOTICE    Nameserver i.ns.se has an IP address (2001:67c:1010:5::53) with mismatched PTR result (se1.dnsnode.net.).

$ # Unsetting LANGUAGE and setting French gives correct language:
$ (unset LANGUAGE; zonemaster-cli iis.se --locale fr_FR.utf8)
Durée   Niveau    Message
======= ========= =======
   5.19 NOTICE    Le serveur de noms i.ns.se a une adresse IP (194.146.106.22) qui ne correspond pas aux enregistrements "PTR" retournés (se1.dnsnode.net.) pour celle-ci.
   5.20 NOTICE    Le serveur de noms i.ns.se a une adresse IP (2001:67c:1010:5::53) qui ne correspond pas aux enregistrements "PTR" retournés (se1.dnsnode.net.) pour celle-ci.
```

**Comment by mattias-p** [comment]
The --locale parameter currently overrides the LC_MESSAGES environment variable, which has medium precedence among internationalization environment variables.  ([POSIX](http://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html), [GNU](https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html)).

Either --locale should be given the highest precedence of all, or it should be replaced entirely with an entry about locale variables in the --help output.

Since LANGUAGE is a GNU extension the fix needs testning on both Linux and FreeBSD.

**Comment by sandoche2k** [comment]
May relate to https://github.com/dotse/zonemaster-backend/issues/315

**Comment by sandoche2k** [comment]
Works for me under Debian 7

```
@vps323915:~# zonemaster-cli zonemaster.net
Seconds Level     Message
======= ========= =======
   1.90 NOTICE    Nameserver ns.nic.se/2001:67c:124c:100a::45 did not respond to NS query.
   1.95 NOTICE    Nameserver ns2.nic.fr/2001:660:3005:1::1:2 did not respond to NS query.
   1.95 NOTICE    Nameserver ns3.nic.se/2001:67c:124c:2007::45 did not respond to NS query.


@vps323915:~# zonemaster-cli zonemaster.net --locale fr_FR.ISO-8859-1
Durée   Niveau    Message
======= ========= =======
   1.89 NOTICE    Le serveur de noms ns.nic.se/2001:67c:124c:100a::45 ne répond pas à une requête de type "NS".
   1.94 NOTICE    Le serveur de noms ns2.nic.fr/2001:660:3005:1::1:2 ne répond pas à une requête de type "NS".
   1.95 NOTICE    Le serveur de noms ns3.nic.se/2001:67c:124c:2007::45 ne répond pas à une requête de type "NS"

```

**Comment by matsduf** [comment]
@sandoche2k, is LANGUAGE set? Please run

```
env | grep -i utf-8;echo;echo $LANGUAGE;echo;zonemaster-cli zonemaster.net --locale fr_FR.ISO-8859-1
```

**Comment by sandoche2k** [comment]
```
env | grep -i utf-8;echo;echo $LANGUAGE;echo;zonemaster-cli zonemaster.net --locale fr_FR.ISO-8859-1
LANG=en_GB.UTF-8



Durée   Niveau    Message
======= ========= =======
   2.02 NOTICE    Le serveur de noms ns.nic.se/2001:67c:124c:100a::45 ne répond pas à une requête de type "NS".
   2.07 NOTICE    Le serveur de noms


```

**Comment by matsduf** [comment]
To see the problem run the following two commands instead:

```
export LANGUAGE=en_GB.UTF-8
zonemaster-cli zonemaster.net --locale fr_FR.ISO-8859-1
```

From the description:
> If the environment variable `LANGUAGE` is set, then the `--locale` option might not work.

**Comment by mattias-p** [comment]
Let --locale override both LC_MESSAGES and LANGUAGE.

**Comment by sandoche2k** [comment]
@vlevigneron will it be possible for looking at this bug in 2018.1?

**Comment by sandoche2k** [comment]
Refer to https://github.com/dotse/zonemaster-cli/pull/71

**Comment by sandoche2k** [comment]
To be closed after testing

**Comment by matsduf** [comment]
@sandoche2k, I think we can close this issue since we have merged #71.
